### ISSUE TITLE
Error when saving SMTP credentials on macOS Sierra: "Error accessing the AutoPkgr keychain"

### ISSUE THREAD

**Issue by homebysix** [lindegroup/autopkgr#554]
The following error appears when attempting to save SMTP credentials in AutoPkgr's Notifications tab on macOS Sierra:

> __Error accessing the AutoPkgr keychain__
> There was a problem accessing the application's keychain and it needs to be recreated.
> This only relates to AutoPkgr's keychain and will not modify or alter any other keychains.

Screenshot:

<img width="532" alt="screen shot 2016-09-19 at 10 05 31 am" src="https://cloud.githubusercontent.com/assets/7801391/18641295/db54c8dc-7e50-11e6-963b-281420781e68.png">

Console output:

```
default	10:10:02.960746 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.960823 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.961020 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.961108 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.961296 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.961401 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.961517 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.961640 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.961738 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.961839 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.961965 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.962052 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.965695 -0700	com.lindegroup.AutoPkgr.helper	0x7fbaacd037b0 free /Library/Keychains/System.keychain buffer 0x7fbaae825c00
default	10:10:02.976596 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.976749 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.976947 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.977036 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.977253 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.977333 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.977499 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.977584 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.977787 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.977871 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.978018 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.978163 -0700	com.lindegroup.AutoPkgr.helper	CSSM Exception: -2147411889 CSSMERR_CL_UNKNOWN_TAG
default	10:10:02.979403 -0700	com.lindegroup.AutoPkgr.helper	not switching as we're not in ~/Library/Keychains/: /Library/Keychains/System.keychain (0)
default	10:10:02.979442 -0700	com.lindegroup.AutoPkgr.helper	not switching as we're not in ~/Library/Keychains/: /Library/Keychains/System.keychain (0)
default	10:10:02.979728 -0700	com.lindegroup.AutoPkgr.helper	DbOpen of /private/var/db/mds/system/mdsObject.db
default	10:10:02.979884 -0700	com.lindegroup.AutoPkgr.helper	0x7fbaace0a7f0 opened /private/var/db/mds/system/mdsObject.db: 4636 bytes
default	10:10:02.980019 -0700	com.lindegroup.AutoPkgr.helper	0x7fbaace0a7f0 allocated /private/var/db/mds/system/mdsObject.db buffer 0x7fbaae006800 size 4636
default	10:10:02.980080 -0700	com.lindegroup.AutoPkgr.helper	0x7fbaace0a7f0 closed /private/var/db/mds/system/mdsObject.db
default	10:10:02.980516 -0700	com.lindegroup.AutoPkgr.helper	DbOpen of /private/var/db/mds/system/mdsDirectory.db
default	10:10:02.980678 -0700	com.lindegroup.AutoPkgr.helper	0x7fbaace0a7f0 free /private/var/db/mds/system/mdsObject.db buffer 0x7fbaae006800
default	10:10:02.980872 -0700	com.lindegroup.AutoPkgr.helper	DbOpen of /Library/Keychains/System.keychain
default	10:10:02.980985 -0700	com.lindegroup.AutoPkgr.helper	0x7fbaacc39ea0 opened /Library/Keychains/System.keychain: 35372 bytes
default	10:10:02.981095 -0700	com.lindegroup.AutoPkgr.helper	0x7fbaacc39ea0 allocated /Library/Keychains/System.keychain buffer 0x7fbaad806000 size 35372
default	10:10:02.981157 -0700	com.lindegroup.AutoPkgr.helper	0x7fbaacc39ea0 closed /Library/Keychains/System.keychain
default	10:10:02.982857 -0700	com.lindegroup.AutoPkgr.helper	loading /Library/Keychains/System.keychain
default	10:10:02.985504 -0700	com.lindegroup.AutoPkgr.helper	not attempting migration for /Library/Keychains/System.keychain version 256 (0 0 1)
default	10:10:02.985737 -0700	com.lindegroup.AutoPkgr.helper	keychain blob version does not support integrity
default	10:10:02.985847 -0700	com.lindegroup.AutoPkgr.helper	skipping integrity check due to keychain version
default	10:10:03.999838 -0700	AutoPkgr	not switching as we're not in ~/Library/Keychains/: /Library/Keychains/System.keychain (0)
default	10:10:03.004252 -0700	com.lindegroup.AutoPkgr.helper	not switching as we're not in ~/Library/Keychains/: /Library/Keychains/System.keychain (0)
default	10:10:03.018885 -0700	AutoPkgr	switching to keychain-db: /Users/linde/Library/Keychains/AutoPkgr.keychain-db from /Users/linde/Library/Keychains/AutoPkgr.keychain (1 1 0 1)
default	10:10:03.018940 -0700	AutoPkgr	switching to keychain-db: /Users/linde/Library/Keychains/AutoPkgr.keychain-db from /Users/linde/Library/Keychains/AutoPkgr.keychain-db (0 1 1 1)
default	10:10:03.019769 -0700	AutoPkgr	DbOpen of /var/folders/4w/sm7fv1cj3mjbr2p5p4z2b2sr0000gn/C//mds/mdsObject.db
default	10:10:03.019878 -0700	AutoPkgr	0x608001275ec0 opened /var/folders/4w/sm7fv1cj3mjbr2p5p4z2b2sr0000gn/C//mds/mdsObject.db: 4636 bytes
default	10:10:03.019927 -0700	AutoPkgr	0x608001275ec0 allocated /var/folders/4w/sm7fv1cj3mjbr2p5p4z2b2sr0000gn/C//mds/mdsObject.db buffer 0x7f88918f6000 size 4636
default	10:10:03.019970 -0700	AutoPkgr	0x608001275ec0 closed /var/folders/4w/sm7fv1cj3mjbr2p5p4z2b2sr0000gn/C//mds/mdsObject.db
default	10:10:03.020353 -0700	AutoPkgr	DbOpen of /var/folders/4w/sm7fv1cj3mjbr2p5p4z2b2sr0000gn/C//mds/mdsDirectory.db
default	10:10:03.020496 -0700	AutoPkgr	0x608001275ec0 free /var/folders/4w/sm7fv1cj3mjbr2p5p4z2b2sr0000gn/C//mds/mdsObject.db buffer 0x7f88918f6000
default	10:10:03.020659 -0700	AutoPkgr	DbCreate of /Users/linde/Library/Keychains/AutoPkgr.keychain-db
default	10:10:03.020749 -0700	AutoPkgr	open /Users/linde/Library/Keychains/AutoPkgr.keychain-db: File exists
default	10:10:03.020816 -0700	AutoPkgr	CSSM Exception: -2147413736 CSSMERR_DL_DATASTORE_ALREADY_EXISTS
default	10:10:03.020937 -0700	AutoPkgr	CSSM Exception: -2147413736 CSSMERR_DL_DATASTORE_ALREADY_EXISTS
default	10:10:03.021026 -0700	AutoPkgr	CSSM Exception: -2147413736 CSSMERR_DL_DATASTORE_ALREADY_EXISTS
```

**Comment by eahrold** [comment]
Should be fixed in this branch https://github.com/lindegroup/autopkgr/tree/feature/Sierra-Fixes

**Comment by homebysix** [comment]
Included in 1.4.2, just released.
